daq/faucet_states_collector.py

```diff
@@ -108,6 +111,82 @@ def get_switch(self, switch_name):
 
         return switch_map
 
+    def get_active_host_route(self, src_mac, dst_mac):
+        """give two MAC addresses in the core network, find the active route between them"""
+        def get_switches_ports_from_link(link_map):
```

Comment:
do we need the nested functions?  It makes the code very much more difficult to read.  Any reason these can't just be methods at the top-level of the class?

Reply:
I think the inner function is only useful for the outter function, i.e., get_active_host_route. If moved ouside as class method, not sure if they will be used for other class methods.

Reply:
Actually I moved out the functions and they are class methods now.